src/Core/src/Layouts/DensityValue.cs

```diff
@@ -0,0 +1,201 @@
+			{
+				throw new ArgumentException("Cannot add DensityValues with different densities.");
+			}
+
+			return DensityValue.FromPixels(left.RawPx + right.RawPx, left.Density);
+		}
+
+		/// <summary>
+		/// Subtracts two DensityValue instances.
+		/// </summary>
+		public static DensityValue operator -(DensityValue left, DensityValue right)
+		{
+			if (Math.Abs(left.Density - right.Density) > Epsilon)
+			{
+				throw new ArgumentException("Cannot subtract DensityValues with different densities.");
+			}
+
+			return DensityValue.FromPixels(left.RawPx - right.RawPx, left.Density);
+		}
+
+		/// <summary>
+		/// Multiplies a DensityValue by a scalar.
+		/// </summary>
+		public static DensityValue operator *(DensityValue value, double scalar)
+		{
+			return DensityValue.FromPixels(value.RawPx * scalar, value.Density);
+		}
+
+		/// <summary>
+		/// Multiplies a DensityValue by a scalar.
+		/// </summary>
+		public static DensityValue operator *(double scalar, DensityValue value)
+		{
+			return value * scalar;
+		}
+
+		/// <summary>
+		/// Divides a DensityValue by a scalar.
+		/// </summary>
+		public static DensityValue operator /(DensityValue value, double scalar)
+		{
+			return DensityValue.FromPixels(value.RawPx / scalar, value.Density);
+		}
+
+		/// <summary>
+		/// Implicitly converts a DensityValue to its dp value.
+		/// </summary>
+		public static implicit operator double(DensityValue value)
+		{
+			return value.Dp;
+		}
+
+		/// <summary>
+		/// Distributes a total pixel amount across multiple DensityValue instances,
+		/// accumulating rounding errors and applying them to the final elements.
+		/// This implements Android's approach of assigning remainder pixels to the last element.
+		/// </summary>
+		/// <param name="totalPixels">The total pixels to distribute.</param>
+		/// <param name="density">The display density.</param>
+		/// <param name="portions">The relative portions for each element.</param>
+		/// <returns>An array of pixel values that sum exactly to totalPixels.</returns>
+		public static int[] DistributePixels(double totalPixels, double density, double[] portions)
+		{
+			if (portions.Length == 0)
+				return Array.Empty<int>();
+
+			var totalPortions = 0.0;
+			foreach (var portion in portions)
+			{
+				totalPortions += portion;
+			}
+
+			if (totalPortions <= 0)
+				return new int[portions.Length]; // All zeros
+
+			var result = new int[portions.Length];
+			var targetTotal = (int)Math.Round(totalPixels);
+			var assignedTotal = 0;
+
+			// Calculate ideal pixels per portion
+			var idealPixelsPerUnit = totalPixels / totalPortions;
+
+			// Assign pixels to all but the last element using floor
+			for (int i = 0; i < portions.Length - 1; i++)
+			{
+				var idealPixels = idealPixelsPerUnit * portions[i];
+				result[i] = (int)Math.Floor(idealPixels);
+				assignedTotal += result[i];
+			}
+
+			// Assign remaining pixels to the last element (this accumulates rounding error)
```

Comment:
Change the implementation of DensityValue so that it distributes the extra pixels from Right to left intead of all onto the last element.

For example,

If the width of a container is 293dp
With a density value of 2.75
And there are nine elements the resulting widths should be

89,89,89,89,89,90,90,90,90